crates/wasmtime/src/runtime/vm/const_expr.rs

```diff
@@ -269,7 +283,19 @@ impl ConstExprEvaluator {
                     let elem = Val::_from_raw(&mut store, self.pop()?, ty.element_type().unpack());
 
                     let pre = ArrayRefPre::_new(&mut store, ty);
-                    let array = ArrayRef::_new(&mut store, &pre, &elem, len)?;
+                    let array = match ArrayRef::_new(&mut store, &pre, &elem, len) {
```

Comment:
Ditto from libcalls, this is being duplicated enough that it should be centralized somehow to a smaller handful of functions than are in use today

Reply:
See my other comment about folding the retry-after-gc into the `FooRef` constructors.